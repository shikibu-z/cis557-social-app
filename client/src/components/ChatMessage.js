/* eslint-disable */
import React from 'react';
import '../stylesheets/ChatMessage.css';

function ChatMessage(props) {

  const { message, author, curUser, attachment } = props;

  const mediaTypes = {
    jpg: 'image',
    png: 'image',
    gif: 'image',
    mp4: 'video',
    '3gp': 'video',
    mp3: 'audio',
    ogg: 'audio',
  };

  let mediaType = '';
  const extension = attachment.split('.').pop();
  if (extension in mediaTypes) {
    mediaType = mediaTypes[extension];
  } else {
    mediaType = 'none';
  }

  if (author === curUser) {
    return (
      <div class="curUserMessage">
        <div class="right-message">
          {mediaType === 'video'
            && (
              <video width="420" height="340" controls>
                <source src={attachment} />
                <track src="captions_en.vtt" kind="captions" label="english_captions" />
              </video>
            )}
          {mediaType === 'image'
            && (
              <img width="200" height="auto" src={attachment} alt="could not media" />
            )}
          {mediaType === 'audio'
            && (
              <audio controls>
                <source src={attachment} />
                <track src="captions_en.vtt" kind="captions" label="english_captions" />
              </audio>
            )}
          {mediaType === 'none'
            && (
              <span>{message}</span>
            )}
        </div>
        {/* <img src="https://picsum.photos/200/300" alt="" /> */}
      </div>
    )
  }

  return (
    <div class="Message">
      {/* <img src='https://source.unsplash.com/ZHvM3XIOHoE' alt="" /> */}
      <div class="left-message">
        {mediaType === 'video'
          && (
            <video width="420" height="340" controls>
              <source src={attachment} />
              <track src="captions_en.vtt" kind="captions" label="english_captions" />
            </video>
          )}
        {mediaType === 'image'
          && (
            <img width="200" height="auto" src={attachment} alt="could not media" />
          )}
        {mediaType === 'audio'
          && (
            <audio controls>
              <source src={attachment} />
              <track src="captions_en.vtt" kind="captions" label="english_captions" />
            </audio>
          )}
        {mediaType === 'none'
          && (
            <span>{message}</span>
          )}
      </div>
    </div>
  );
}

export default ChatMessage;
