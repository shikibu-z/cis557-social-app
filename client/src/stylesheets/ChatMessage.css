.curUserMessage {
  display: flex;
  flex-direction: row;
  margin:2% 2% 2% auto;
  background: #F5F5F7;
  border-radius: 10px;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: 2%;
}

.right-message span {
  vertical-align: middle;
  font-size: 1.2em;
}

/* .curUserMessage img {
  width:60px;
  height:60px;
  border-radius: 50%;
  overflow:hidden;
  margin: auto 1% auto 4%;
} */


.Message {
  display: flex;
  flex-direction: row;
  margin: 1% auto 1% 1%;
  background: #F5F5F7;
  border-radius: 10px;
  display: flex;
  align-items: center;
  justify-content: center;
} 

/* .Message img {
  width:60px;
  height:60px;
  border-radius: 50%;
  overflow:hidden;
  margin: auto 1% auto 4%;
} */

/* .left-message {
  background: #F5F5F7;
  border-radius: 10px;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-left: 2%;
} */

.left-message span {
  vertical-align: middle;
  font-size: 1.2em;
}