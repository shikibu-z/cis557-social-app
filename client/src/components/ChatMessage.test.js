/*eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import ChatMessage from './ChatMessage';

test('ChatMessage matches snapshot', () => {
  const component = renderer.create(<ChatMessage message="" author="" curUser="" attachment="test.jpeg" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
