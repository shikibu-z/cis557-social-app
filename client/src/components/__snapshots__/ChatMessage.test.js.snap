// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessage matches snapshot 1`] = `
<div
  class="curUserMessage"
>
  <div
    class="right-message"
  >
    <span>
      
    </span>
  </div>
</div>
`;
